tiledb/api/c_api/attribute/attribute_api.cc

```diff
@@ -135,7 +135,28 @@ int32_t tiledb_attribute_get_cell_size(
 int32_t tiledb_attribute_dump(
     const tiledb_attribute_handle_t* attr, FILE* out) {
   ensure_attribute_is_valid(attr);
-  attr->dump(out);
+  ensure_cstream_handle_is_valid(out);
+
+  std::stringstream ss;
+  ss << *attr;
```

Comment:
```suggestion
  ss << attr->attr_;
```

You don't have to declare a separate `operator <<` for each handle (same with the other types).

Reply:
`attribute_type attr_` is private member.

Reply:
>attribute_type attr_ is private member

It's private by design. It needs to stay that way. C API implementation functions should not have direct access to the underlying object. Instead, they should access it only through facade interfaces. This is in the design document in the `c_api` directory.